src/Tribe/REST/V1/Post_Repository.php

```diff
@@ -0,0 +1,408 @@
+<?php
+
+
+class Tribe__Events__REST__V1__Post_Repository implements Tribe__Events__REST__Interfaces__Post_Repository {
+
+	/**
+	 * A post type to get data request handler map.
+	 *
+	 * @var array
+	 */
+	protected $types_get_map = array();
+
+	/**
+	 * @var Tribe__REST__Messages_Interface
+	 */
+	protected $messages;
+
+	public function __construct( Tribe__REST__Messages_Interface $messages = null ) {
+		$this->types_get_map = array(
+			Tribe__Events__Main::POSTTYPE            => array( $this, 'get_event_data' ),
+			Tribe__Events__Main::VENUE_POST_TYPE     => array( $this, 'get_venue_data' ),
+			Tribe__Events__Main::ORGANIZER_POST_TYPE => array( $this, 'get_organizer_data' ),
+		);
+
+		$this->messages = $messages ? $messages : tribe( 'tec.rest-v1.messages' );
+	}
+
+	/**
+	 * Retrieves an array representation of the post.
+	 *
+	 * @param int $id The post ID.
+	 *
+	 * @return array An array representation of the post.
+	 */
+	public function get_data( $id ) {
+		$post = get_post( $id );
+
+		if ( empty( $post ) ) {
+			return array();
+		}
+
+		if ( ! isset( $this->types_get_map[ $post->post_type ] ) ) {
+			return (array) $post;
+		}
+
+		return call_user_func( $this->types_get_map[ $post->post_type ], $id );
+	}
+
+	/**
+	 * Returns an array representation of an event.
+	 *
+	 * @param int $event_id An event post ID.
+	 *
+	 * @return array|WP_Error Either the array representation of an event or an error object.
+	 */
+	public function get_event_data( $event_id ) {
+		$event = get_post( $event_id );
+
+		if ( empty( $event ) || ! tribe_is_event( $event ) ) {
+			return new WP_Error( 'event-not-found', $this->messages->get_message( 'event-not-found' ) );
+		}
+
+		$meta = array_map( 'reset', get_post_custom( $event_id ) );
+
+		$venue = $this->get_venue_data( $event_id );
+		$organizer = $this->get_organizer_data( $event_id );
```

Comment:
Since the venue and organizer stuff is hard-coded rather than being dynamically attached to the data, we probably should add a ticket in Central for the future addition of "Linked Post Types" support in our JSON-ED.